master/static/srv/get_experiments.sql

```diff
@@ -1,15 +1,33 @@
+WITH filtered_exps AS (
+    SELECT
+        e.id AS id,
+        e.config->>'description' AS description,
+        e.config->'labels' AS labels,
+        e.config->'resources'->>'resource_pool' AS resource_pool,
+        e.start_time AS start_time,
+        e.end_time AS end_time,
+        'STATE_' || e.state AS state,
+        (SELECT COUNT(*) FROM trials t WHERE e.id = t.experiment_id) AS num_trials,
+        e.archived AS archived,
+        COALESCE(e.progress, 0) AS progress,
+        u.username AS username
+    FROM experiments e
+    JOIN users u ON e.owner_id = u.id
+    WHERE
+        ($1 = '' OR e.state IN (SELECT unnest(string_to_array($1, ','))::experiment_state))
+        AND ($2 = '' OR e.archived = $2::BOOL)
+        AND ($3 = '' OR (u.username IN (SELECT unnest(string_to_array($3, ',')))))
+        AND (
+                $4 = ''
+                OR string_to_array($4, ',') <@ ARRAY(SELECT jsonb_array_elements_text(e.config->'labels'))
+            )
```

Comment:
I think this is probably going to be a little rougher than the other filters because of the computation (`jsonb_array_elements_text`) on the filtered field. If you can think of any way to avoid that @shiyuann, it might be good? We should definitely watch the DB at release party to see how this query is hitting the load so we can be ready to optimize it more pre-release if needed.

Reply:
`jsonb_array_elements_text` only expands a JSON array to a set of text values. I don't think it would be the place where the DB CPU will be spent the most. The query itself only goes through the whole temporary table to get the count as opposed to marshaling the whole table. Both the DB and the master should have lower pressure unless the experiments are very few.

Reply:
All I meant was `jsonb_array_elements_text` was the only expression on the RHS of a filter so it will have to be performed for any rows that haven't been removed yet.

Reply:
Even if we change the column to be an array type, it is still linear complexity to the length of the column, I think